source/libraries/ContractInterfacesGenerator.ts

```diff
@@ -0,0 +1,181 @@
+     */
+
+    export class Contract {
+        protected readonly connector: Connector;
+        protected readonly accountManager: AccountManager;
+        public readonly address: string;
+        protected readonly defaultGasPrice: BN;
+
+        protected constructor(connector: Connector, accountManager: AccountManager, address: string, defaultGasPrice: BN) {
+            this.connector = connector;
+            this.accountManager = accountManager;
+            this.address = address;
+            this.defaultGasPrice = defaultGasPrice;
+        }
+
+        protected async localCall(abi: AbiFunction, parameters: Array<any>, sender?: string, attachedEth?: BN): Promise<Array<any>> {
+            const from = sender || this.accountManager.defaultAddress;
+            const data = encodeMethod(abi, parameters);
+            const transaction = Object.assign({ from: from, to: this.address, data: data }, attachedEth ? { value: attachedEth } : {});
+            const result = await this.connector.ethjsQuery.call(transaction);
+            return decodeParams(abi.outputs.map(x => x.name), abi.outputs.map(x => x.type), result);
+        }
+
+        protected async remoteCall(abi: AbiFunction, parameters: Array<any>, sender?: string, gasPrice?: BN, attachedEth?: BN): Promise<string> {
+            const from = sender || this.accountManager.defaultAddress;
+            const data = encodeMethod(abi, parameters);
+            // TODO: remove \`gas\` property once https://github.com/ethereumjs/testrpc/issues/411 is fixed
+            const gas = await this.connector.ethjsQuery.estimateGas(Object.assign({ to: this.address, from: from, data: data, gas: new BN(6500000) }, attachedEth ? { value: attachedEth } : {} ));
+            gasPrice = gasPrice || this.defaultGasPrice;
+            const transaction = Object.assign({ from: from, to: this.address, data: data, gasPrice: gasPrice, gas: gas }, attachedEth ? { value: attachedEth } : {});
+            const signedTransaction = await this.accountManager.signTransaction(transaction);
+            return await this.connector.ethjsQuery.sendRawTransaction(signedTransaction);
+        }
+    }
+
+    ${contractInterfaces.join("\n")}
+    export function ContractFactory(connector: Connector, accountManager: AccountManager, address: string, defaultGasPrice: BN): Controlled {
+        return new Controlled(connector, accountManager, address, defaultGasPrice);
+    }
+    `;
+}
+
+    function contractInterfaceTemplate(contractName: String, contractAbi: Abi) {
```

Comment:
Indentation is off from here down I believe.  This should be at the root (unless you wrap the whole thing in a class).